app/controllers/RelayRound.scala

```diff
@@ -0,0 +1,209 @@
+                rt => JsonOk(env.relay.jsonView.admin(rt.relay))
+              )
+          }
+    )
+
+  private def doUpdate(id: String, me: UserModel)(implicit
+      req: Request[_]
+  ): Fu[Option[Either[(RoundModel.WithTour, Form[RelayRoundForm.Data]), RoundModel.WithTour]]] =
+    env.relay.api.byIdAndContributor(id, me) flatMap {
+      _ ?? { rt =>
+        env.relay.roundForm
+          .edit(rt.relay)
+          .bindFromRequest()
+          .fold(
+            err => fuccess(Left(rt -> err)),
+            data =>
+              env.relay.api.update(rt.relay) { data.update(_, me) }.dmap(_ withTour rt.tour) dmap Right.apply
+          ) dmap some
+      }
+    }
+
+  def reset(id: String) =
+    Auth { implicit ctx => me =>
+      OptionFuResult(env.relay.api.byIdAndContributor(id, me)) { rt =>
+        env.relay.api.reset(rt.relay, me) inject Redirect(rt.path)
+      }
+    }
+
+  def show(ts: String, rs: String, id: String) =
+    OpenOrScoped(_.Study.Read)(
+      open = implicit ctx => {
+        pageHit
+        WithRelay(ts, rs, id) { rt =>
+          val sc =
+            if (rt.relay.sync.ongoing)
+              env.study.chapterRepo relaysAndTagsByStudyId rt.relay.studyId flatMap { chapters =>
+                chapters.find(_.looksAlive) orElse chapters.headOption match {
+                  case Some(chapter) => env.study.api.byIdWithChapter(rt.relay.studyId, chapter.id)
+                  case None          => env.study.api byIdWithChapter rt.relay.studyId
+                }
+              }
+            else env.study.api byIdWithChapter rt.relay.studyId
+          sc flatMap { _ ?? { doShow(rt, _) } }
+        }
+      },
+      scoped = _ =>
+        me =>
+          env.relay.api.byIdAndContributor(id, me) map {
+            case None     => NotFound(jsonError("No such broadcast"))
+            case Some(rt) => JsonOk(env.relay.jsonView.admin(rt.relay))
+          }
+    )
+
+  def chapter(ts: String, rs: String, id: String, chapterId: String) =
+    Open { implicit ctx =>
+      WithRelay(ts, rs, id) { rt =>
+        env.study.api.byIdWithChapter(rt.relay.studyId, chapterId) flatMap {
+          _ ?? { doShow(rt, _) }
+        }
+      }
+    }
+
+  def cloneRelay(id: String) =
+    Auth { implicit ctx => me =>
+      OptionFuResult(env.relay.api.byIdAndContributor(id, me)) { rt =>
+        env.relay.api.cloneRelay(rt, me) map { newRelay =>
+          Redirect(routes.RelayRound.edit(newRelay.id.value))
+        }
+      }
+    }
+
+  def push(id: String) =
+    ScopedBody(parse.tolerantText)(Seq(_.Study.Write)) { req => me =>
+      env.relay.api.byIdAndContributor(id, me) flatMap {
+        case None     => notFoundJson()
+        case Some(rt) => env.relay.push(rt, req.body) inject jsonOkResult
+      }
+    }
+
+  // def apiIndex =
+  //   Action.async { implicit req =>
+  //     apiC.jsonStream {
+  //       env.relay.api.officialStream(MaxPerSecond(20), getInt("nb", req) | 20)
+  //     }.fuccess
+  //   }
```

Comment:
Can the commented code be cleaned up?

Reply:
Please stop. This is obviously still marked WIP